src/environment.rs

```diff
@@ -8,26 +8,38 @@
 //! Unlike its siblings, this module should be considered private and only used
 //! via the re-exports one level up.
 
+pub mod mutex;
+
+use crate::abi::GuestRet;
+
+use crate::environment::mutex::{host_mutex_is_locked, host_mutex_relock_unblocked};
```

Comment:
Nit: might want to consolidate this with the `use self::mutex::` below and just write `use mutex::`. Also remove the double new lines here.